### PR TITLE
pstress-69 Introduce new option exact-initial-records

### PR DESCRIPTION
**Pull request by rahulmalik87**
Insert exact number of initial records mention with --records
Otherwise it randomly insert records in each table

### CODE REVIEW

File: src/random_test.cpp [hunk 1]
```diff
@@ -1709,7 +1709,9 @@ void load_default_data(Table *table, Thd1 *thd) {
   thd->ddl_query = false;
   static int initial_records =
       options->at(Option::INITIAL_RECORDS_IN_TABLE)->getInt();
-  int rec = rand_int(initial_records);
+  int rec = options->at(Option::EXACT_INITIAL_RECORDS)->getBool()
```
Comment (mohitj1988) [inline src/random_test.cpp:1712]: What happens if a user passes --jldll. The load_default_data() will not be called and no records would be inserted in tables

File: src/random_test.cpp [hunk 2]
```diff
@@ -1709,7 +1709,9 @@ void load_default_data(Table *table, Thd1 *thd) {
...
+                ? initial_records
+                : rand_int(initial_records);
   for (int i = 0; i < rec; i++) {
     table->InsertRandomRow(thd, false);
```
Comment (mohitj1988) [inline src/random_test.cpp:1716]: The inserted number of rows in the table does not match the provided number of records. For eg
./pstress-ps --tables 1 --logdir=`pwd`/log --threads 10 --socket /tmp/mysql_22000.sock --seconds 1 --records 20 --exact-initial-records
This inserts 
mysql> select count(*) from tt_1_p;
+----------+
| count(*) |
+----------+
|       27 |
+----------+
1 row in set (0.00 sec)

mysql> select count(*) from tt_1;
+----------+
| count(*) |
+----------+
|       35 |
+----------+
1 row in set (0.01 sec)

File: src/help.cpp [hunk 3]
```diff
@@ -251,6 +251,14 @@ void add_options() {
   opt->help = "Number of initial records in table";
   opt->setInt(1000);
 
+  /* Initial Records in table */
+  opt = newOption(Option::BOOL, Option::EXACT_INITIAL_RECORDS,
+                  "exact-initial-records");
+  opt->help = "Insert exact number of initial records mention with --records. "
+              "Otherwise it randomly insert records in each table";
+  opt->setBool(false);
+  opt->setArgs(no_argument);
```
Comment (mohitj1988) [inline src/help.cpp:260]: The new option --exact-initial-records must not accept any argument. It does at the moment. Please check.

File: src/random_test.cpp (hunk 1 above)
Comment (rahulmalik87) [reply src/random_test.cpp:1712]: jlddl is to created ddl and exit.

File: src/random_test.cpp (hunk 2 above)
Comment (rahulmalik87) [reply src/random_test.cpp:1716]: seconds 1 mean load is executed for 1 seconds.  try with --seconds 0. there may be 10% less becuase of some failure of DML due to unique/primary index but not huge difference

File: src/random_test.cpp (hunk 1 above)
Comment (mohitj1988) [reply src/random_test.cpp:1712]: Ok. You can ignore this comment.

File: src/random_test.cpp (hunk 2 above)
Comment (mohitj1988) [reply src/random_test.cpp:1716]: mohit.joshi@qaserver-06:~/pstress_8.0/src$ ./pstress-ps --tables 1 --logdir=`pwd`/log --threads 10 --socket /tmp/mysql_22000.sock --seconds 0 --records 20 --exact-initial-records
- Connecting to default.node.tld []...
metadata created randomly
The value of rec:20
The value of rec:20
starting random load in 10 threads.
Saving metadata to file /home/mohit.joshi/pstress_8.0/src/log/step_1.dll
COMPLTED
mysql> select count(*) from tt_1;
+----------+
| count(*) |
+----------+
|       20 |
+----------+
1 row in set (0.00 sec)

mysql> select count(*) from tt_1_p;
+----------+
| count(*) |
+----------+
|       15 |
+----------+
1 row in set (0.00 sec)

Comment (mohitj1988) [reply src/random_test.cpp:1716]: For --tables 1 --records 200, --exact-initial-records
select count(*) from tt_1_p;    80
select count(*) from tt_1;   199

File: src/help.cpp (hunk 3 above)
Comment (rahulmalik87) [reply src/help.cpp:260]: ./pstress-ms  --socket $SOCKET --table 4 --records 100 --no-table-compression --exact-initial-records=1 --seconds 100 --log-all-queries
pstress-ms: option `--exact-initial-records' doesn't allow an argument
Invalid option , exiting
{Rahuls-MacBook-Pro x28 o8 src}$

File: src/random_test.cpp (hunk 2 above)
Comment (rahulmalik87) [reply src/random_test.cpp:1716]: tt_1_p is list partition. that's fine. we can't expect lot of enties in them.

Comment (rahulmalik87) [reply src/random_test.cpp:1716]: Error Table has no partition for value 66.

File: src/help.cpp (hunk 3 above)
Comment (mohitj1988) [reply src/help.cpp:260]: ./pstress-ps --tables 1 --logdir=`pwd`/log --threads 10 --socket /tmp/mysql_22000.sock --seconds 0 --records 200 --exact-initial-records 1000
- Connecting to default.node.tld []...
metadata created randomly
The value of rec:200
The value of rec:200
starting random load in 10 threads.
Saving metadata to file /home/mohit.joshi/pstress_8.0/src/log/step_1.dll
COMPLTED

Run without "="

Comment (rahulmalik87) [reply src/help.cpp:260]: all other binary options are also like that. for e.g check --jlddl. the behaviour is fine.

File: src/random_test.cpp (hunk 2 above)
Comment (mohitj1988) [reply src/random_test.cpp:1716]: Ok

File: src/help.cpp [hunk 4]
```diff
@@ -251,6 +251,14 @@ void add_options() {
   opt->help = "Number of initial records in table";
```
Comment (mohitj1988) [inline src/help.cpp:251]: Please change this:
opt->help = "initial records in table are inserted randomly in the range of 0 to N";

File: src/help.cpp [hunk 5]
```diff
@@ -251,6 +251,14 @@ void add_options() {
...
   opt->setInt(1000);
 
+  /* Initial Records in table */
+  opt = newOption(Option::BOOL, Option::EXACT_INITIAL_RECORDS,
+                  "exact-initial-records");
+  opt->help = "Insert exact number of initial records mention with --records. "
```
Comment (mohitj1988) [inline src/help.cpp:257]: opt->help = "--exact-initial-records when passed with --records option inserts"
                   " exact number of records in tables"

**Pull request merged**